azerothcore/azerothcore-wotlk
# fix(Core/Player) Fix Rediscovering Certain Flight Paths and Zones After a Logout and Login

Certain zones and flight path discoveries are removed on logout and must
be frustratingly rediscovered again

Closes Azerothcore issue #7737



## Changes Proposed:
-  It apparently fixes the conversion of strings to unsigned longs.  Certain zones and flight paths must not convert right so they do not save on a logout.  
-  Got the info from [https://github.com/TrinityCore/TrinityCore/commit/24c173f387044496d581255545ed94c988f3ae3b](url)
and [https://github.com/TrinityCore/TrinityCore/issues/13493](url)
## Issues Addressed:

- Closes #7737

## SOURCE:


## Tests Performed:

- Went the to flight masters specified in the Issues (Nutral, Bladruc, Celea) and checked to see if flight the flight was saved on logout.
- Learned spells and recipes and generally tried to break stuff.
- Went to zones like thelsamar and 
- Compiles and runs just fine.

## How to Test the Changes:


1. Go to Baldruc in Theramore and learn his flight path.  Logout and log back in and see if it still is learned/discovered
2. Go to Valley of Fangs in Badlands and discover it then logout and log back and see if you rediscover it.  
3.

## Known Issues and TODO List:

Please test this thing as I am not a great programmer. 


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
